### ISSUE TITLE
Retina @2X Image not loaded in retina resolution

### ISSUE THREAD

**Issue by fortunecreations** [imulus/retinajs#220]
Hello,
I hope doing well.
I have use ratina.js file in my theme. all thing is okay.
Retina working successfully,his create retina image. see screenshot : http://prntscr.com/8w86tf
but I have use feature image and get that this type in my home page template file. see screenshot : http://prntscr.com/8w8785
and
I have test my theme is retina ready or not using this method. 
See Screenshot : http://prntscr.com/8w8785
URL : http://stackoverflow.com/questions/15551287/how-to-test-a-website-for-retina-on-windows-without-an-actual-retina-display
but in retina resolution @2x image not load. same image load in normal screen.
please help me :)

Thank you

**Comment by majorgilles** [comment]
Yes, the library only allows for specific characters in the filename, hence the issues you are encountering.

the issue is in the $filename_pattern = '[ \w\\.\\/\\-]+';  variable in ImageServiceProviderLaravel5 service provider. Adding \\@ will solve the problem.

**Comment by jamescalhoun** [comment]
Hi,

I have been having similar, but more erratic, issues with images not appearing on a site that I have inherited.  I noticed that the site was having some issues with images occasionally not displaying on retina devices. I assumed this was an implementation issue, as I have seen Retina.js work well before.  Perhaps this could be related to the issue discussed here https://github.com/imulus/retinajs/issues/8 ?

It seems that all of the images which were affected had slightly (1px) difference in sizing between the base and the @2x version.

Not sure this is helpful, but just wanted to share what I came across. When I get the varnish caching issue cleared up on production I will let you know if this helped :)

-James

**Comment by ramedia** [comment]
Make sure there are no animation or transition properties on the image or parent div's.
I had the problem of the images not showing retina pictures in Safari, but they were fine on desktop browsers.

Maybe that can help. Another option is conflicting js library's, so try loading it after or before pageload in javascript.
